src/main/java/life/qbic/data/processing/Provenance.java

```diff
@@ -0,0 +1,111 @@
+package life.qbic.data.processing;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * <b>Provenance Information of Datasets</b>
+ * <p>
+ * Captures some dataset provenance metadata for pre-processing purposes.
+ *
+ * @since 1.0.0
+ */
+@JsonIgnoreProperties
+public class Provenance {
+
+  public static final String FILE_NAME = "provenance.json";
+
+  /**
+   * The path from where the dataset has been picked up originally.
+   */
+  @JsonProperty("origin")
+  public String originPath;
+
+  @JsonProperty("user")
+  public String userPath;
```

Comment:
what path is this?

Reply:
Yes exactly, the user's working directory, if you want. I try to find a more suitable semantic